File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q2-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let result = [];
    for (let i = 0; i < numRows; i++) {
        let row = [];
        row[0] = 1;
        if (i > 0) {
            row[i] = result[i - 1][i - 1] + result[i - 1][i];
            for (let j = 1; j < i; j++) {
                row[j] = result[i - 1][j - 1] + result[i - 1][j];
            }
        }
        result.push(row);
    }
    return result;
};

File: ELEC881_Copilot/src/Original Prompt/Python/q2-pascals-triangle.py
'''
Given an integer numRows, return the first numRows of Pascal's triangle.

In Pascal's triangle, each number is the sum of the two numbers directly above it as shown:


 

Example 1:

Input: numRows = 5
Output: [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]
Example 2:

Input: numRows = 1
Output: [[1]]
 

Constraints:

1 <= numRows <= 30
'''
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if numRows == 1:
            return [[1]]
        elif numRows == 2:
            return [[1], [1, 1]]
        else:
            res = [[1], [1, 1]]
            for i in range(2, numRows):
                res.append([1])
                for j in range(1, i):
                    res[i].append(res[i-1][j-1] + res[i-1][j])
                res[i].append(1)
            return res